examples/applications/plot_time_series_lagged_features.py

```diff
@@ -0,0 +1,395 @@
+
+# %%
+# We can compute several combinations of evaluation metrics and loss functions,
+# which are reported a bit below.
+from collections import defaultdict
+
+from sklearn.metrics import (
+    make_scorer,
+    mean_absolute_error,
+    mean_pinball_loss,
+    root_mean_squared_error,
+)
+from sklearn.model_selection import cross_validate
+
+scoring = {
+    "MAPE": make_scorer(mean_absolute_percentage_error),
+    "RMSE": make_scorer(root_mean_squared_error),
+    "MAE": make_scorer(mean_absolute_error),
+    "pinball_loss_05": make_scorer(mean_pinball_loss, alpha=0.05),
+    "pinball_loss_50": make_scorer(mean_pinball_loss, alpha=0.50),
+    "pinball_loss_95": make_scorer(mean_pinball_loss, alpha=0.95),
+}
+loss_functions = ["squared_error", "poisson", "absolute_error"]
+scores = defaultdict(list)
+for loss_func in loss_functions:
+    model = HistGradientBoostingRegressor(loss=loss_func)
+    cv_results = cross_validate(
+        model,
+        X,
+        y,
+        cv=ts_cv,
+        scoring=scoring,
+    )
+    time = cv_results["fit_time"]
+    scores["loss"].append(loss_func)
+    scores["fit_time"].append(f"{time.mean():.2f} ± {time.std():.2f} s")
+
+    for key, value in cv_results.items():
+        if key.startswith("test_"):
+            metric = key.split("test_")[1]
+            scores[metric].append(f"{value.mean():.3f} ± {value.std():.3f}")
+
+
+# %%
+# Modeling Predictive Uncertainty via Quantile Regression
+# -------------------------------------------------------
+# Instead of modeling the expected value of the distribution of
+# :math:`Y|X` like the least squares and Poisson losses do, one could try to
+# estimate quantiles of the conditional distribution.
+#
+# :math:`Y|X=x_i` is expected to be a random variable for a given data point
+# :math:`x_i` because we expect that the number of rentals cannot be 100%
+# accurately predicted from the features. It can be influenced by other
+# variables not properly captured by the existing lagged features. For
+# instance whether or not it will rain in the next hour cannot be fully
+# anticipated from the past hours bike rental data. This is what we
+# call aleatoric uncertainty.
+#
+# Quantile regression makes it possible to give a finer description of that
+# distribution without making strong assumptions on its shape.
+quantile_list = [0.05, 0.5, 0.95]
+
+for quantile in quantile_list:
+    model = HistGradientBoostingRegressor(loss="quantile", quantile=quantile)
+    cv_results = cross_validate(
+        model,
+        X,
+        y,
+        cv=ts_cv,
+        scoring=scoring,
+    )
+    time = cv_results["fit_time"]
+    scores["fit_time"].append(f"{time.mean():.2f} ± {time.std():.2f} s")
+    scores["loss"].append(f"quantile {int(quantile*100)}")
+    for key, value in cv_results.items():
+        if key.startswith("test_"):
+            metric = key.split("test_")[1]
+            scores[metric].append(f"{value.mean():.3f} ± {value.std():.3f}")

```

Comment:
The same change should be applied here. I thing you should factorize the post processing of the `cv_results` in a `consolidate_scores(cv_results, scores)` function to reuse twice.